tests/ui/lint/dangling-pointers-from-temporaries/types.rs

```diff
@@ -52,6 +52,14 @@ fn main() {
     //~^ ERROR a dangling pointer will be produced because the temporary `UnsafeCell<u8>` will be dropped
     declval::<SyncUnsafeCell<u8>>().get();
     //~^ ERROR a dangling pointer will be produced because the temporary `SyncUnsafeCell<u8>` will be dropped
-    declval::<Box<AsPtrFake>>().as_ptr();
+    declval::<MaybeUninit<MaybeUninit<&u8>>>().as_ptr();
+    //~^ ERROR a dangling pointer will be produced because the temporary `MaybeUninit<MaybeUninit<&u8>>` will be dropped
+    declval::<Box<MaybeUninit<&u8>>>().as_ptr();
+    //~^ ERROR a dangling pointer will be produced because the temporary `Box<MaybeUninit<&u8>>` will be dropped
+
+    // should not lint
+    declval::<&[u8]>().as_ptr();
     declval::<AsPtrFake>().as_ptr();
+    declval::<Box<AsPtrFake>>().as_ptr();
+    declval::<MaybeUninit<&u8>>().as_ptr();
```

Comment:
But this one *should* lint?

`MaybeUninit<&u8>.as_ptr()` will return a `*const &u8` which does not point to a heap allocation, true, but it still points to memory owned by the MaybeUninit (it points inside of the MaybeUninit union itself). So if it is a temporary, then this pointer will dangle.

Reply:
Indeed, good find. Fixed it.